workshop_app/views.py

```diff
@@ -44,23 +44,39 @@ def is_instructor(user):
     return user.groups.filter(name='instructor').exists()
 
 
-def get_landing_page(user):
-    # For now, landing pages of both instructor and coordinator are same
+def instructor_only(func):
+    def inner1(*args, **kwargs):
+        if is_instructor(args[0].user):
+            return func(*args, **kwargs)
+        return redirect(reverse('workshop_status_coordinator'))
+
+    return inner1
+
+
+def coordinator_only(func):
+    def inner1(*args, **kwargs):
+        if not is_instructor(args[0].user):
+            return func(*args, **kwargs)
+        return redirect(reverse('workshop_status_instructor'))
+
```

Comment:
For these decorators create a separate file called decorators.py. This should not be done in views.